cloud/scope/machine.go

```diff
@@ -1083,8 +1083,6 @@ func (m *MachineScope) SetProviderID(id *string) error {
 			return err
 		}
 		m.IBMVPCMachine.Spec.ProviderID = ptr.To(fmt.Sprintf("ibm://%s///%s/%s", accountID, m.Machine.Spec.ClusterName, *id))
-	} else {
```

Comment:
I think in main.go we have a check to make sure only v2 provider id is valid, So couple of things here
1. Is if check at 1079 needed?
2. If we keep if condition then if its not v2 format we should return error.

Reply:
Fixed.